### PR TITLE
(#458) Prevent crashes with non-UTF-8 files

### PR DESCRIPTION
**Pull request by rnelson0**
A hopefully simpler replacement for #459

### CODE REVIEW

**Comment by rnelson0** [comment]
The failure is on ruby 1.8.7, which presumably does not throw an ArgumentError. I'm not certain how to conditionally test that for 1.8.7, or if it really matters with plans to phase that out before 3.0.0.

File: lib/puppet-lint/bin.rb [hunk 1]
```diff
@@ -74,6 +74,9 @@ def run
       puts "puppet-lint: no file specified or specified file does not exist"
       puts "puppet-lint: try 'puppet-lint --help' for more information"
       return 1
+    rescue ArgumentError
```
Comment (binford2k) [inline lib/puppet-lint/bin.rb]: What's raising the error? Are we depending on an implementation detail? Is this going to mask other errors?

ArgumentError seems overly broad.

**Comment by rnelson0** [comment]
Revise: use `force_encoding("ISO-8859-1").encode("utf-8", replace: nil).` to drop non-UTF-8 characters, in the 3.x branch.

**Comment by rodjek** [comment]
Closing this out as it this approach shouldn't be necessary with #730

**Pull request closed without merging**